src/main/scala/beam/cosim/helics/BeamFederate.scala

```diff
@@ -23,11 +23,14 @@ object BeamFederate {
   var beamFed = Option.empty[BeamFederate]
 
   def getInstance(beamServices: BeamServices): BeamFederate = this.synchronized {
-    if (beamFed.isEmpty) {
-      loadHelics
-      beamFed = Some(BeamFederate(beamServices))
+    beamFed match {
+      case Some(beamFederate) => beamFederate
```

Comment:
this method looks like buggy. pls double check because `beamFed` is changed only in the second case

Reply:
its working like this: If beamFed is initialized then return its instance otherwise initialize it then reurn its instance. So second case is basically handle its initialization